lib/scnnr/connection.rb

```diff
@@ -31,22 +34,42 @@ def send_json(data)
     def send_request
       block = block_given? ? Proc.new : nil
       request = build_request(&block)
-      run_request(request)
+      with_retries do
+        Net::HTTP.start(@uri.host, @uri.port, use_ssl: use_ssl?) do |http|
+          @logger&.info("Started #{@method.upcase} #{@uri}")
+          http.request(request)
+        end
+      end
     end
 
     private
 
-    def run_request(request)
-      Net::HTTP.start(@uri.host, @uri.port, use_ssl: use_ssl?) do |http|
-        @logger&.info("Started #{@method.upcase} #{@uri}")
-        http.request(request)
+    def with_retries
+      yield
+    rescue Errno::ECONNRESET => e
+      retry_count ||= 0
+
+      if retry_count < RETRY_LIMIT
+        retry_count += 1
+        @logger&.info("Retrying to connect: #{@uri}, attempt: #{retry_count}")
+
+        sleep RETRY_SLEEP_TIME
+        retry
       end
+
+      raise RequestFailed, error_attributes(e)
```

Comment:
Currently, we defined `RequestFailed` as an error that represents "an request to get an image specified by a user is failed.".
In this case, should we wrap the actual error? I think it would be enough to re-raise it.

Reply:
You're right, I've changed it to re-raise the error!